application/src/ext/java/org/opentripplanner/ext/fares/impl/OrcaFareService.java

```diff
@@ -348,11 +362,22 @@ private Optional<Money> getSeniorFare(
     Leg leg
   ) {
     var route = leg.getRoute();
-    var regularFare = getRegularFare(fareType, rideType, defaultFare, leg);
+    if (route == null) {
+      return Optional.of(defaultFare);
+    }
     // Many agencies only provide senior discount if using ORCA
     return switch (rideType) {
-      case COMM_TRANS_LOCAL_SWIFT -> optionalUSD(1.25f);
-      case COMM_TRANS_COMMUTER_EXPRESS -> optionalUSD(2f);
+      case COMM_TRANS_LOCAL_SWIFT -> {
+        if (
+          leg
+            .getStartTime()
+            .isBefore(LocalDate.of(2025, 3, 1).atStartOfDay(leg.getStartTime().getZone()))
```

Comment:
Extract this into a constant, like a `private static final` field and use it above as well. Actually the two pieces of code are identical so extract a method for it.

Reply:
also added a todo to remove all this in march